### PR TITLE
#115 C++ api

### PR DESCRIPTION
**Pull request by SanderMertens**

### CODE REVIEW

File: include/flecs/flecs.hpp [hunk 1]
```diff
@@ -0,0 +1,1571 @@
...
+        ecs_assert(index < m_count, ECS_COLUMN_INDEX_OUT_OF_RANGE, NULL);
+        return m_array[index];
+    }
+
+    T* operator->() {
+        ecs_assert(m_array != nullptr, ECS_COLUMN_INDEX_OUT_OF_RANGE, NULL);
+        return m_array;
+    }
+
+    bool is_set() {
```
Comment (Milerius) [inline include/flecs/flecs.hpp]: this function could be marked as cont

File: include/flecs/flecs.hpp [hunk 2]
```diff
@@ -0,0 +1,1571 @@
...
+static const entity_t Singleton = EcsSingleton;
+static const entity_t Invalid = EcsInvalid;
+
+
+////////////////////////////////////////////////////////////////////////////////
+//// Wrapper class around a table column
+////////////////////////////////////////////////////////////////////////////////
+
+template <typename T>
+class column final {
```
Comment (Milerius) [inline include/flecs/flecs.hpp]: Why the column is marked as final, without inheritance ?

File: include/flecs/flecs.hpp [hunk 3]
```diff
@@ -0,0 +1,1571 @@
...
+public:
+    column(T* array, size_t count, bool is_shared = false)
```
Comment (Milerius) [inline include/flecs/flecs.hpp]: You should prefer using `std::size_t` for c++ version of C type, (<cstddef>)

File: include/flecs/flecs.hpp [hunk 4]
```diff
@@ -0,0 +1,1571 @@
...
+    T* operator->() {
+        ecs_assert(m_array != nullptr, ECS_COLUMN_INDEX_OUT_OF_RANGE, NULL);
+        return m_array;
+    }
+
+    bool is_set() {
+        return m_array != nullptr;
+    }
+
+    bool is_shared() {
```
Comment (Milerius) [inline include/flecs/flecs.hpp]: Can also be const

File: include/flecs/flecs.hpp [hunk 5]
```diff
@@ -0,0 +1,1571 @@
...
+    type table_type() const;
+
+    /* Obtain untyped pointer to table column */
+    void* table_column(uint32_t table_column) {
+        return ecs_table_column(m_rows, table_column);
+    }
+
+    /* Obtain typed pointer to table column */
+    template <typename T>
+    flecs::column<T> table_column() {
```
Comment (Milerius) [inline include/flecs/flecs.hpp]: I think this function can be marked as const

File: include/flecs/flecs.hpp (hunk 2 above)
Comment (SanderMertens) [reply include/flecs/flecs.hpp]: I prevent inheriting from framework classes wherever possible, just because it's bad practice.

**Pull request merged**